### PR TITLE
Enhance search bar UX

### PR DESCRIPTION
**Pull request by danielcolinjames**
- addresses #735
- add loading state
- add "No results" state
- only allow pressing Enter after loading results is complete

### CODE REVIEW

**Comment by cokes518** [comment]
expand the search box, add padding to the address so it doesn't appear behind the icon.

image

**Comment by danielcolinjames** [comment]
added some nice improvements for mobile:


and for desktop:

**Comment by cokes518** [comment]
🎈
________________________________

File: components/SearchBar/SearchBar.js [hunk 1]
```diff
@@ -102,45 +115,93 @@ const SearchBar = () => {
...
+            ref={scroll}
+            className="absolute bg-white max-h-96 md:max-h-72 md:w-96 left-2 md:left-auto right-2 lg:right-4 top-14 rounded-lg divide-y divide-gray-400 overflow-y-scroll no-scrollbar shadow-md z-40"
+          >
+            {/* show that results are loading once you start typing */}
+            {resultsLoading ? (
+              <BaseSearchResult
+                title="Loading results..."
+                subtitle="Talking to the API..."
+              />
+            ) : results.length === 0 ? (
```
Comment (allenan) [inline components/SearchBar/SearchBar.js]: nested ternaries are a bit of an antipattern. could we extract a component out of here where we could then use traditional logical operators?

File: components/SearchBar/useSearchResults.js [hunk 2]
```diff
@@ -222,27 +238,29 @@ const useSearchResults = () => {
...
         // transaction hash
-        searchTransaction(term)
+        await searchTransaction(term)
       } else {
-        searchHotspot(term.replace(/-/g, ' '))
-        searchValidator(term.replace(/-/g, ' '))
-        // searchCities(term)
-        searchMapAddresses(term)
-        searchMaker(term)
+        await searchHotspot(term.replace(/-/g, ' '))
```
Comment (allenan) [inline components/SearchBar/useSearchResults.js]: this is going to perform each of these searches sequentially. is that what we want?

**Comment by allenan** [comment]
really nice ux enhancement for mobile 🚀

File: components/SearchBar/SearchBar.js (hunk 1 above)
Comment (danielcolinjames) [reply components/SearchBar/SearchBar.js]: good call 👍

File: components/SearchBar/useSearchResults.js (hunk 2 above)
Comment (danielcolinjames) [reply components/SearchBar/useSearchResults.js]: good catch, wrapped in a Promise.all

**Review by allenan — APPROVED** [review]
revisions look good

**Pull request merged**